ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -1571,6 +1693,10 @@ static void ggml_vk_load_shaders(vk_device& device) {
                                               uint32_t parameter_count, uint32_t push_constant_size, std::array<uint32_t, 3> wg_denoms, const std::vector<uint32_t>& specialization_constants,
                                               uint32_t align, bool disable_robustness = false, bool require_full_subgroups = false, uint32_t required_subgroup_size = 0) {
 
+        if (!require_full_subgroups) {
```

Comment:
One minor issue, like this it's impossible to use the `required_subgroup_size`, since it gets overwritten here. That might make sense, but then the parameter could also be removed. Or we only override it if it is equal to 0.

Reply:
You're right, I don't know when would the parameter be used but I just think the best way is to override it if it's not 0, just to keep the original behavior of the program.